src/cache-restore.ts

```
@@ -0,0 +1,55 @@
+import * as cache from '@actions/cache';
+import * as core from '@actions/core';
+import path from 'path';
+import fs from 'fs';
+
+import {State, Outputs} from './constants';
+import {
+  getCacheDirectoryPath,
+  hashFile,
+  isPackageManagerCacheSupported
+} from './cache-utils';
+
+export const restoreCache = async (packageManager: string, version: string) => {
+  if (!isPackageManagerCacheSupported(packageManager)) {
+    throw new Error(`Caching for '${packageManager}'is not supported`);
+  }
+  const lockKey = findLockFile(packageManager);
+
+  const platform = process.env.RUNNER_OS;
+  const fileHash = await hashFile(lockKey);
+
+  const primaryKey = `${platform}-${packageManager}-${version}-${fileHash}`;
+  core.saveState(State.CachePrimaryKey, primaryKey);
+
+  const cachePath = await getCacheDirectoryPath(packageManager);
+  const cacheKey = await cache.restoreCache([cachePath], primaryKey);
+
+  if (!cacheKey) {
+    core.warning(`${packageManager} cache is not found`);
```

Do we really want to throw warning every time when cache is not found? It means that customers will see warning every time when they update `package-lock.json`. Let's do it `core.info`

---

Done. Thank you.